src/pyhf/pdf.py

```diff
@@ -731,9 +731,10 @@ def logpdf(self, pars, data):
             ):  # force to be not scalar, should we changed with #522
                 return tensorlib.reshape(result, (1,))
             return result
-        except:  # noqa: E722
+        except Exception as excep:
             log.error(
-                f'eval failed for data {tensorlib.tolist(data)} pars: {tensorlib.tolist(pars)}'
+                repr(excep)
+                + f'\nEval failed for data {tensorlib.tolist(data)} pars: {tensorlib.tolist(pars)}'
```

Comment:
@kratsg has pointed out that this can also just be done with [`log.error(..., exc_info=True)`](https://docs.python.org/3/library/logging.html#logging.Logger.error) which gets info from [`sys.exc_info`](https://docs.python.org/3/library/sys.html#sys.exc_info).